taichi/program/program.cpp

```diff
@@ -448,8 +442,11 @@ SNode *Program::get_snode_root(int tree_id) {
 void Program::materialize_snode_tree(SNodeTree *tree) {
   auto *const root = tree->root();
   // always use host_arch() for host accessors
+  std::unique_ptr<llvm::Module> host_module = nullptr;
+  if (!snode_trees_.empty())
+    host_module = llvm_context_host->clone_struct_module();
```

Comment:
It was a bit confusing to me how come this didn't crash on the first call, then I realized that this might be calling `clone_runtime_module()` inside the constructor.

nit: I believe we can make this a bit more obvious ;-) E.g.

```cpp
... host_module = nullptr;
if (!...)
  host_module = llvm...->clone_struct_module();
if (!host_module)
  host_module = llvm...->clone_runtime_module();
```
Then we can get rid of the `module == nullptr ? ... : ...` inside `StructCompilerLLVM`.